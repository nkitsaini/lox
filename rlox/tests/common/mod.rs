// These are macros otherwise insta snapshot goes crazy.
// It starts having different folder locations for `cargo insta test` and `cargo test`

macro_rules! test_execution {
    ($result:expr, $script:expr) => {{
        let mut stdout = std::io::Cursor::new(Vec::new());
        let mut stderr = std::io::Cursor::new(Vec::new());
        let mut vm = rlox::vm::VM::empty_new(&mut stdout, &mut stderr);
        let res = vm.interpret($script);

        drop(vm);

        pretty_assertions::assert_eq!(res, $result);
        insta::assert_snapshot!(String::from_utf8(stderr.into_inner()).unwrap());
        insta::assert_snapshot!(String::from_utf8(stdout.into_inner()).unwrap());
    }};
}
macro_rules! test_execution_success {
    ($script:ident) => {
        test_execution!(Ok(()), $script);
    };
}
