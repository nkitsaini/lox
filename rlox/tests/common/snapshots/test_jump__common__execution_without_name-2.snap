---
source: tests/common/mod.rs
expression: "String::from_utf8(stdout.into_inner()).unwrap()"
---
0
1
3

